index.js

```diff
@@ -47,8 +47,32 @@ const isObject = value => value !== null && typeof value === 'object';
 const supportsAbortController = typeof globals.AbortController === 'function';
 const supportsStreams = typeof globals.ReadableStream === 'function';
 
+const mergeHeaders = (...sources) => {
+	const result = {};
+
+	for (const source of sources) {
+		if (!isObject(source)) {
+			throw new TypeError('The `headers` argument must be an object');
+		}
+
+		const headers = new globals.Headers(source);
```

Comment:
This line is unnecessary

Reply:
Oh, actually it is necessary since the input can be a Request instance, and `request.headers` is an instance of `Headers` :+1: